docs/topics/settings.rst

```diff
@@ -98,6 +98,26 @@ and settings set there should use the "spider" priority explicitly:
             super().update_settings(settings)
             settings.set("SOME_SETTING", "some value", priority="spider")
 
+.. versionadded:: VERSION
+
+It's also possible to modify the settings in the
+:meth:`~scrapy.Spider.from_crawler` method, e.g. based on :ref:`spider
+arguments <spiderargs>` or other logic:
+
+.. code-block:: python
+
+    import scrapy
+
+
+    class MySpider(scrapy.Spider):
+        name = "myspider"
+
+        @classmethod
+        def from_crawler(cls, crawler, *args, **kwargs):
+            spider = super().from_crawler(crawler, *args, **kwargs)
+            spider.settings.set("SOME_SETTING", kwargs["value"], priority="spider")
```

Comment:
Since it would only be 1 more line, I wonder if the example should ensure `kwargs["value"]` actually exists before using it, to encourage best practices.